### PR TITLE
chore(filewriter): cleanup writes

### PR DESCRIPTION
**Pull request by schomatis**

### CODE REVIEW

**Comment by dud1337** [comment]
Anything holding up merging this?

**Comment by dud1337** [comment]
Is there a reason this isn't merged?

**Review by aschmahmann — APPROVED** [review]

**Comment by BigLep** [comment]
@aschmahmann @schomatis : can we merge?

File: filewriter.go [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+		return ErrPathExistsOverwrite
+	} else if !os.IsNotExist(err) {
+		return err
+	}
 	switch nd := nd.(type) {
 	case *Symlink:
 		return os.Symlink(nd.Target, fpath)
 	case File:
-		f, err := os.Create(fpath)
+		f, err := os.OpenFile(fpath, os.O_EXCL|os.O_CREATE|os.O_WRONLY, 0666)
```
Comment (aschmahmann) [inline filewriter.go:25]: @schomatis could you add back the O_NOFOLLOW flag for when we're running on systems that support it? We already have some OS specific files here so it should be pretty easy to add in another function to get OS specific symlink flags.

**Review by aschmahmann — APPROVED** [review]
Thanks 🙏

**Pull request merged**